### PR TITLE
Ability to provide TF commit to CMake as well as clearing of CMake cache variables #30

### PR DESCRIPTION
**Pull request by cpetig**
rebased

### CODE REVIEW

**Comment by cpetig** [comment]
I feel with the tests passing we have little to loose ;-)

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,18 +36,17 @@ ENDIF()
 
 SET(TF_LIB tensorflow-microlite)
 
+FILE(GLOB COMPILER_HEADERS ${PROJECT_SOURCE_DIR}/src/*.h)
```
Comment (rafzi) [inline CMakeLists.txt:39]: globbing source files in cmake should generally be avoided: https://cmake.org/cmake/help/v3.15/command/file.html?highlight=glob#file

File: CMakeLists.txt [hunk 2]
```diff
@@ -66,6 +65,16 @@ TARGET_COMPILE_DEFINITIONS(${PROJECT_NAME} PUBLIC
...
+ADD_CUSTOM_TARGET(
+    format
+    COMMAND clang-format -i ${COMPILER_SRCS} ${COMPILER_HEADERS}
+    )
+
 IF(TF_EXAMPLES)
     ADD_SUBDIRECTORY(${PROJECT_SOURCE_DIR}/examples)
 ENDIF()
+
+UNSET(TF_EXAMPLES CACHE)
```
Comment (rafzi) [inline CMakeLists.txt:77]: this is pretty unconventional. if i set a setting in cmake, i expect it to stick until i change it again or clear the cache myself

Comment (alxhoff) [reply CMakeLists.txt:77]: Ah yes the examples should probably stay set. Given the constant need to change the fetched TF code base (tag, commit etc) it made sense to me that each call I could provide a different TF code base that I wanted.

File: CMakeLists.txt (hunk 1 above)
Comment (alxhoff) [reply CMakeLists.txt:39]: you're making me be less lazy

**Pull request merged**